lib/execution_engine2/utils/job_requirements_resolver.py

```diff
@@ -0,0 +1,166 @@
+"""
+Contains resolvers for job requirements.
+"""
+
+from typing import Dict, Union
+
+from execution_engine2.sdk.job_submission_parameters import JobRequirements
+from execution_engine2.exceptions import IncorrectParamsException
+
+CLIENT_GROUP = "client_group"
+REQUEST_CPUS = "request_cpus"
+REQUEST_MEMORY = "request_memory"
+REQUEST_DISK = "request_disk"
+CLIENT_GROUP_REGEX = "client_group_regex"
+DEBUG_MODE = "debug_mode"
+_RESOURCES = set([CLIENT_GROUP, REQUEST_CPUS, REQUEST_MEMORY, REQUEST_DISK])
+
+
+def _check_raise(name, value, source):
+    raise IncorrectParamsException(
+        f"Found illegal {name} '{value}' in job requirements from {source}"
+    )
+
+
+def _check_clientgroup(clientgroup, source):
+    clientgroup = _string_request(clientgroup, "client group", source)
+    # this is a possible error mode from the catalog since it uses key=value pairs in CSV
+    # format
+    if "=" in clientgroup:
+        _check_raise("client group", clientgroup, source)
+    return clientgroup
+
+
+def _string_request(putative_string, name, source):
+    if type(putative_string) != str:
+        _check_raise(name, putative_string, source)
+    return putative_string.strip()
+
+
+def _int_request(putative_int, original, name, source):
+    if type(putative_int) == float:
+        _check_raise(f"{name} request", original, source)
+    try:
+        return int(putative_int)
+    except ValueError:
+        _check_raise(f"{name} request", original, source)
+
+
+def _check_cpus(cpus, source):
+    return _int_request(cpus, cpus, "cpu", source)
+
+
+def _check_memory(memory, source):
+    if type(memory) == int:
+        return memory
+    memory2 = _string_request(memory, "memory request", source)
+    if memory2.endswith("M"):
+        memory2 = memory2[:-1]
+    elif memory2.endswith("MB"):
+        memory2 = memory2[:-2]
```

Comment:
are those the only options?

Reply:
(i.e. no GB / TB / KB / etc.)

Reply:
Currently yes, the three sources of job requirements are:
- the run as concierge method, which takes ints
- the catalog, which currently only specifies numbers as strings with no units
- https://github.com/kbase/execution_engine2/blob/develop/deploy.cfg, which is `M` (and `GB` for disk space)